hibernate-core/src/main/java/org/hibernate/persister/collection/AbstractCollectionPersister.java

```diff
@@ -2183,6 +2186,9 @@ public Object getElementByIndex(Object key, Object index, SharedSessionContractI
 
 	@Override
 	public boolean isExtraLazy() {
+		if ( isArray ) {
+			return false;
+		}
 		return isExtraLazy;
 	}
```

Comment:
Maybe creating an `AbstractArrayPersister` is a better idea. Effective Java warns the bad practice to rely on such flag field to  create different logic.